cmd/controller/diskstore.go

```diff
@@ -124,13 +124,14 @@ func (c *Controller) GetExistingResource(listDr *apis.DiskList, uuid string) *ap
 // list of active resources. One active resource which is present in etcd not in
 // system that will be marked as inactive.
 func (c *Controller) DeactivateStaleDiskResource(devices []string) {
+	listDisks := append(devices, GetActiveSparseDisksUuids(c.HostName)...)
```

Comment:
this looks OK, but it is supposed to be part of scan function, which is passing devices to the  DeactivateStaleDiskResource. As Scan function's job is to scan every resource that NDM is managing, it could be either block device or sparse file.